knowledge_engine/knowledge_engine/conversion/local_markdown.py

```
@@ -0,0 +1,244 @@
+
+
+def _convert_epub(binary_data: bytes) -> str:
+    if not zipfile.is_zipfile(BytesIO(binary_data)):
+        raise RuntimeError("Invalid EPUB file: expected a ZIP package")
+
+    with zipfile.ZipFile(BytesIO(binary_data)) as archive:
+        rootfile = _find_epub_rootfile(archive)
+        if not rootfile:
+            raise RuntimeError("Invalid EPUB file: missing OPF rootfile")
+
+        opf_root = ElementTree.fromstring(archive.read(rootfile))
```

Ruff (0.15.21)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)